numba/dispatcher.py

```diff
@@ -659,6 +674,11 @@ def __init__(self, py_func, locals={}, targetoptions={},
         functools.update_wrapper(self, py_func)
 
         self.targetoptions = targetoptions
+        if "disable_reflected_list" in py_func.__globals__ and \
```

Comment:
@stuartarchibald you mentioned you have a solution for comparing by identity here?

Reply:
it would look like this:
```python
from numba.future import disable_reflected_list
if any(disable_reflected_list is x for x in py_func.__globals__.values()):
```

Reply:
ah yes, apologies, just saw this.